nilearn/_utils/numpy_conversions.py

```diff
@@ -130,3 +130,29 @@ def csv_to_array(csv_path, delimiters=" \t,;", **kwargs):
         )
 
     return array
+
+
+def get_target_dtype(dtype, target_dtype):
+    """Return a new dtype if conversion is needed.
+
+    Parameters
+    ----------
+    dtype : dtype
+        Data type of the original data
+
+    target_dtype : {None, dtype, "auto"}
+        If None, no conversion is required. If a type is provided, the
+        function will check if a conversion is needed. The "auto" mode will
+        automatically convert to int32 if dtype is discrete and float32 if it
+        is continuous.
+
+    Returns
+    -------
+    dtype : dtype
+        The data type toward which the original data should be converted.
+    """
+    if target_dtype is None:
+        return None
+    if target_dtype == "auto":
+        target_dtype = np.int32 if dtype.kind == "i" else np.float32
```

Comment:
```suggestion
        target_dtype = 'int32' if dtype.kind == "i" else 'float32'
``` (I thought we rather use core Python dtypes ?)

Reply:
fixing: should not make a big difference though